### ISSUE TITLE
A components referenced in `components` and also used as a typeof will lead to an error with `"importsNotUsedAsValues": "error"`

### ISSUE THREAD

**Issue by segevfiner** [qmhc/vite-plugin-dts#209]
### Describe the bug

If you import a component and use it in the `components` option while also using it in a `typeof`, you will get an:
```
This import is never used as a value and must use 'import type' because 'importsNotUsedAsValues' is set to 'error'
```

As components is set to `undefined` and is the only other value reference. 

### Reproduction

https://stackblitz.com/edit/vitejs-vite-dteehr?file=src/components/Bar.vue

### Steps to reproduce

1. Run `npm run build`.

### System Info

```shell
System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 16.14.2 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 7.17.0 - /usr/local/bin/npm
  npmPackages:
    @vitejs/plugin-vue: ^4.2.1 => 4.2.1 
    vite: ^4.3.5 => 4.3.5 
    vite-plugin-dts: ^2.3.0 => 2.3.0
```

### Validations

- [X] Read the [FAQ](https://github.com/qmhc/vite-plugin-dts#faq).
- [X] Check that there isn't [already an issue](https://github.com/qmhc/vite-plugin-dts/issues) that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by segevfiner** [comment]
We are also hitting https://github.com/qmhc/vite-plugin-dts/issues/209

**Comment by qmhc** [comment]
Fixed in `3.0.0-beta.2`.
